waiter/integration/waiter/basic_test.clj

```diff
@@ -1716,3 +1716,103 @@
        (is (contains? details "id"))
        (is (contains? details "name"))
        (is (contains? details "port"))))))
+
+(deftest ^:parallel ^:integration-fast test-signal-sigkill-instance
+  (testing-using-waiter-url
+    (let [
+          headers {:x-waiter-name (rand-name)
+                   :x-waiter-min-instances 2}
+          _ (log/info "making canary request...")
+          {:keys [cookies instance-id service-id]} (make-request-with-debug-info headers #(make-kitchen-request waiter-url %))]
+          ;signal-endpoint (str "/apps/" service-id "/signal/signal-kill/" instance-id)
+      (with-service-cleanup
+        service-id
+        (testing "Call signal sigkill endpoint"
+          (assert-service-on-all-routers waiter-url service-id cookies)
+          ;; assert instance-id is in active instances
+          (let [service-settings (service-settings waiter-url service-id)
+                active-instances (get-in service-settings [:instances :active-instances])
+                num-active-instances (count (get-in service-settings [:instances :active-instances]))
+                active-instance-ids (set (map #(get % :id) active-instances))]
+            (is (<= 1 num-active-instances))
+            (is (contains? active-instance-ids instance-id)))
+          ; call our endpoint and validate response
+          (let [signal-response {:success true :message (str ":sigkill successfully sent to " instance-id) :status 200}]
+                (-> (make-request waiter-url (str "/apps/" service-id "/signal/sigkill/" instance-id) :method :post :query-params {"timeout"  10000})
+                    (assert-response-status http-200-ok))
+                (let [{:keys [body] :as response}
+                      (make-request waiter-url (str "/apps/" service-id "/signal/sigkill/" instance-id) :method :post :query-params {"timeout"  10000})]
+                  (assert-response-status response http-200-ok)
+                  (let [response-data (-> body str try-parse-json walk/keywordize-keys)]
+                    (is (= signal-response (:signal-response response-data)))
+                  ))
+          (is (wait-for #(->> (get-in (service-settings waiter-url service-id) [:instances :killed-instances])
+                              (map :id)
+                              set
+                              seq)
+                        :interval 2 :timeout 1000)
```

Comment:
Is one-second timeout too aggressive?
We should also assert that every router has killed instances.